org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -272,13 +273,13 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
 	 */
 	@Override
 	public void initialized(InitializedParams params) {
-		logInfo(">> initialized");
+		debugTrace(">> initialized");
```

Comment:
Changing the log behavior will have an impact on our langage daemon process in vscode-java-pack extension.

We use a daemon process in vscode-java-pack to monitor and analyse this file. Our daemon relies on the indicator information such as `>> initialized`, >> `initialization job finished`, `>> build jobs finished` to measure the performance of each stage. If the logs are turned off by default, our daemon extension will not work correctly.

In vscode-java, we only write the logs to the file, which does not affect the user experience. I’m curious why the Eclipse client redirects them to the log console instead of the file.

// cc: @Eskibear

Reply:
> I’m curious why the Eclipse client redirects them to the log console instead of the file.

The eclipse client does embed JDT-LS in the workspace, inside the process, sharing the same Eclipse Platform. So whatever in JDT-LS uses the Eclipse API triggers standard handling for this API in Eclipse IDE; ie in Eclipse Platform, every log event produced by Platform.getLog() goes to the log.

I've suggested using the Eclipse debug options, but if it's better, it could be a custom initialization option that could be used to disable logging, opt-in by default and eclipseide-jdtls disabling it.

Reply:
Yes, a custom initialization option could be a good approach to keep compatibility. We have used the custom initialization options a lot in vscode-java, see https://github.com/redhat-developer/vscode-java/blob/fdaacd883537caae81691aa871606933b65f7709/src/extension.ts#L156-L177

Besides, it might be better to keep the lifecycle related logs such as `>> initialized`, `>> initialization job finished`, `>> build jobs finished`, `>> shutdown`, `>> exit`. They're usually helpful.

Reply:
I updated my branch according to this feedback but the PR doesn't update it...

Reply:
Commit is now https://github.com/mickaelistria/eclipse.jdt.ls/commit/4a17a5328aea7293aa1557de5d327f2cd35dfea0